src/Movie.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+
+const Movie = props => {
+  function rocketFavText(id) {
+    // Toggle the text of RocketFave section if that movie is a savedMovie or not
+    const savedMovieIdArr = (props.savedMovies.map(savedMovie => {
+      return savedMovie.id;
+    }));
+    if (savedMovieIdArr.indexOf(id) >= 0) {
+      return 'Remove RocketFave';
+    } else {
+      return 'Add to RocketFaves';
+    }
+  }
+  function checkLength() {
```

Comment:
I like that you are refactoring out some render logic!

Two thoughts, I think you can take this further, and split this logic in to 2 separate functions, I would also name the functions starting with `renderBlahBlah` it's just a pattern but helps easily identify what the functions intention is.